scripts/pg-drop

```diff
@@ -0,0 +1,233 @@
+import sys
+from collections import OrderedDict
+
+import swsssdk
+from natsort import natsorted
+from tabulate import tabulate
+
+# mock the redis for unit test purposes #
+try:
+    if os.environ["UTILITIES_UNIT_TESTING"] == "2":
+        modules_path = os.path.join(os.path.dirname(__file__), "..")
+        tests_path = os.path.join(modules_path, "tests")
+        sys.path.insert(0, modules_path)
+        sys.path.insert(0, tests_path)
+        import mock_tables.dbconnector
+
+except KeyError:
+    pass
+
+STATUS_NA = 'N/A'
+
+COUNTER_TABLE_PREFIX = "COUNTERS:"
+
+COUNTERS_PORT_NAME_MAP = "COUNTERS_PORT_NAME_MAP"
+COUNTERS_PG_NAME_MAP = "COUNTERS_PG_NAME_MAP"
+COUNTERS_PG_PORT_MAP = "COUNTERS_PG_PORT_MAP"
+COUNTERS_PG_INDEX_MAP = "COUNTERS_PG_INDEX_MAP"
+
+def get_dropstat_dir():
+    dropstat_dir_prefix = '/tmp/dropstat'
+    return "{}-{}/".format(dropstat_dir_prefix, os.getuid())
+
+class PGdropstat(object):
+
+    def __init__(self):
+        self.counters_db = swsssdk.SonicV2Connector(host='127.0.0.1')
+        self.counters_db.connect(self.counters_db.COUNTERS_DB)
+
+        dropstat_dir = get_dropstat_dir()
+        self.port_drop_stats_file = os.path.join(dropstat_dir, 'pg_drop_stats')
+
+        def get_pg_port(table_id):
+            port_table_id = self.counters_db.get(self.counters_db.COUNTERS_DB, COUNTERS_PG_PORT_MAP, table_id)
+            if port_table_id is None:
+                print("Port is not available in table '{}'".format(table_id), file=sys.stderr)
+                sys.exit(1)
+
+            return port_table_id
+
+        # Get all ports
+        self.counter_port_name_map = self.counters_db.get_all(self.counters_db.COUNTERS_DB, COUNTERS_PORT_NAME_MAP)
+        if self.counter_port_name_map is None:
+            print("COUNTERS_PORT_NAME_MAP is empty!", file=sys.stderr)
+            sys.exit(1)
+
+        self.port_pg_map = {}
+        self.port_name_map = {}
+
+        for port in self.counter_port_name_map:
+            self.port_pg_map[port] = {}
+            self.port_name_map[self.counter_port_name_map[port]] = port
+
+
+        # Get PGs for each port
+        counter_pg_name_map = self.counters_db.get_all(self.counters_db.COUNTERS_DB, COUNTERS_PG_NAME_MAP)
+        if counter_pg_name_map is None:
+            print("COUNTERS_PG_NAME_MAP is empty!", file=sys.stderr)
+            sys.exit(1)
+
+        for pg in counter_pg_name_map:
+            port = self.port_name_map[get_pg_port(counter_pg_name_map[pg])]
+            self.port_pg_map[port][pg] = counter_pg_name_map[pg]
+
+        self.pg_drop_types = {
+            "pg_drop"       : {"message" : "Ingress dropped packets PG:",
+                               "obj_map" : self.port_pg_map,
+                               "idx_func": self.get_pg_index,
+                               "counter_name" : "SAI_INGRESS_PRIORITY_GROUP_STAT_DROPPED_PACKETS",
+                               "header_prefix": "PG"},
+        }
+
+    def get_pg_index(self, table_id):
```

Comment:
i am not sure i understand the concept of table. do we have more than 1 table for the PG? what are the expected tables?

Reply:
**table_id** - is actually object id  for entry in redis db. 
For example **oid:0x1a000000000121** - is our id
**COUNTERS:oid:0x1a000000000121** - full path for id
**redis-cli -n 2  HGET  COUNTERS:oid:0x1a000000000121 SAI_INGRESS_PRIORITY_GROUP_STAT_DROPPED_PACKETS**- command for retrieving value